### PR TITLE
lib-classifier: Refactor banner components to use react-i18next

### PR DESCRIPTION
**Pull request by goplayoutside3**
# Translations Management
This PR refactors lib-classifer's banner components to use [react-i18next](https://react.i18next.com/) instead of `counterpart` for string translations.

I also integrated `react-i18next` into Storybook, and in the future we'll be able to test locale switching in component stories when other languages' dictionary files are ready.

Translations project board: https://github.com/zooniverse/front-end-monorepo/projects/15

## Package:
lib-classifier

## Describe your changes:
Translation keys are organized into `/src/translations` with two dictionary files per locale. `/en/components.json` contains translations for and is organized in the same manner as each component in `/src/components/Classifier/components`. `/en/plugins` contains translations for `/src/plugins`.

Each component that used `counterpart('Some.String.Here')` now uses the useTranslation() hook as `t('Some.String.Here')`.

Changes to Test Files:
In several cases, `/Banners` test files were explicitly looking for matching strings, and I ended up refactoring them for a few reasons. I had a difficult time getting our test environment to render components with `useTranslation()` strings instead of the translation key. Storybook is a better tool for testing what strings will look like in a browser, and now if we change strings in our dictionary files, we won't need to refactor tests in the future.

### To Test
You can either run `yarn storybook` with lib-classifier, or doublecheck the components render correctly as children of app-project by running `yarn bootstrap` with this branch. Each banner story should display exactly the same when run locally versus our production [storybook](https://zooniverse.github.io/front-end-monorepo/@zooniverse/classifier/index.html?path=/story/banners-alreadyseenbanner--default).

## General
- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 

## Apps
- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Post-merging
- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
- [ ] Is this ready for production deployment?

### CODE REVIEW

File: packages/lib-classifier/src/components/Classifier/components/Banners/components/SubjectSetProgressBanner/SubjectSetProgressBanner.js [hunk 1]
```diff
@@ -20,29 +17,27 @@ function SubjectSetProgressBanner({
...
-  statusText = subject?.retired ? `${counterpart('SubjectSetProgressBanner.finished')}` : statusText
-  const progressText = counterpart('SubjectSetProgressBanner.bannerText', {
-    number: subject?.priority,
-    setName,
-    total: subjectTotal
-  })
-  const tooltipText = counterpart('SubjectSetProgressBanner.tooltipText')
+  let statusText = subject?.alreadySeen ? `${t('Banners.SubjectSetProgressBanner.alreadySeen')}` : ''
+  statusText = subject?.retired ? `${t('Banners.SubjectSetProgressBanner.finished')}` : statusText
+  const progressText = `${t('Banners.SubjectSetProgressBanner.bannerText', { setName })} ${subject?.priority}/${subjectTotal}`
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/components/Classifier/components/Banners/components/SubjectSetProgressBanner/SubjectSetProgressBanner.js:31]: In RTL languages, like Hebrew, we may need to reverse this entire string, not just the output of the translation function.

Comment (goplayoutside3) [reply packages/lib-classifier/src/components/Classifier/components/Banners/components/SubjectSetProgressBanner/SubjectSetProgressBanner.js:31]: Good point! I played around with switching locale in storybook and setting the document's body `dir` to 'rtl'. The previous translation setup works better than having to manually reverse the string in SubjectSetProgressBanner. I'll revert to the old string setup.

Comment (eatyourgreens) [reply packages/lib-classifier/src/components/Classifier/components/Banners/components/SubjectSetProgressBanner/SubjectSetProgressBanner.js:31]: Cool that the storybook lets us test RTL languages. I think Mark has links to at least one live project in Hebrew.

**Comment by goplayoutside3** [comment]
@eatyourgreens re-requesting a review when you have time! Further translations PR's to lib-classifier are blocked by this one as it introduces /en folder of dictionary files.

**Comment by eatyourgreens** [comment]
No problem. I can take another look tomorrow. Happy to approve it as is.

**Review by eatyourgreens — APPROVED** [review]

**Pull request merged**